src/main/java/teammates/ui/controller/InstructorCourseInstructorEditSaveAction.java

```diff
@@ -55,6 +63,25 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         return result;
     }
 
+    /* This method gets the numbers of instructors that would be made visible to students if
+    changes to the instructor information were successfully processed. Thus, if we are
+    trying to make the instructor false, we need to account for that.
+     */
+    protected int getNumberOfInstructorsDisplayedToStudents(String courseId, boolean isDisplayedToStudents) {
+        List<InstructorAttributes> instructors = logic.getInstructorsForCourse(courseId);
+        int numOfInstrDisplayed = 0;
+        for (InstructorAttributes instructor : instructors) {
+            if (instructor.isDisplayedToStudents) {
+                numOfInstrDisplayed++;
+            }
+        }
+        // We need to account for the instructor we're trying to make false.
+        if (!isDisplayedToStudents) {
+            numOfInstrDisplayed--;
+        }
```

Comment:
This check seems to result in a bug: if there are two instructors in a course, one visible and one non-visible, and we attempt to modify the non-visible instructor (perhaps by changing the instructor's permissions for instance), but not modifying the visibility, this will show an error to the user.